pkg/cloudevents/transport/amqp/codec_test.go

```diff
@@ -371,7 +371,7 @@ func TestCodecRoundTrip(t *testing.T) {
 					got.Data = data
 				}
 
-				if tc.wantErr != nil || err != nil {
```

Comment:
This does not catch the case where wantErr is nil and err is not.

Reply:
That can't happen, on line 354 we have if err != nil { ... return } so if we get here err == nil.
That's why the linter is complaining - it's an impossible condition.